plugins/module_utils/dependency_checker.py

```diff
@@ -0,0 +1,160 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) IBM Corporation 2025
+# Licensed under the Apache License, Version 2.0 (the "License");
+# You may not use this file except in compliance with the License.
+# You may obtain a copy at:
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import, division, print_function
+import sys
+from ansible_collections.ibm.ibm_zos_core.plugins.module_utils import version
+try:
+    from zoautil_py import zsystem
+except ImportError:
+    zsystem = None
+
+
+__metaclass__ = type
+
+# ------------------------------------------------------------------------------
+# Compatibility Matrix by Collection Version
+# ------------------------------------------------------------------------------
+# Define compatible versions of Python, ZOAU, and z/OS for each collection version.
+COMPATIBILITY_MATRIX = {
+    "2.0.0": [
+        {"zoau_version": "1.4.0", "min_python_version": "3.12", "max_python_version": "3.13", "min_zos_version": 2.5, "max_zos_version": 3.1},
+        {"zoau_version": "1.4.1", "min_python_version": "3.12", "max_python_version": "3.13", "min_zos_version": 2.5, "max_zos_version": 3.1},
+        {"zoau_version": "1.4.2", "min_python_version": "3.12", "max_python_version": "3.13", "min_zos_version": 2.5, "max_zos_version": 3.1},
+    ],
+    "2.1.0": [
+        {"zoau_version": "1.4.1", "min_python_version": "3.12", "max_python_version": "3.13", "min_zos_version": 2.5, "max_zos_version": 3.1},
```

Comment:
For the version compatibilities of python you can refer to this

<img width="1141" height="852" alt="Image" src="https://github.com/user-attachments/assets/43abb69a-d698-4678-b611-8050f16b4bfc" />

<img width="1218" height="486" alt="Image" src="https://github.com/user-attachments/assets/e0819912-72ae-4863-9765-bde2fce8626a" />

We are currently in AAP 2.3, 2.4 so ansible 2.15-2.18, python version 3.11-3.13 is the current supported version, but this check should be a warning only.